### PR TITLE
PP-5377 Disallow Payment Creation on Invalid Mandate State

### PR DESCRIPTION
**Pull request by RoryMalcolm**
- If a mandate has an invalid state and cannot take payments, currently the payment is accepted then an error occurs further down the line of the user journey.
- This fix changes that and does not allow payments to be created if a mandate has an invalid state, instead an error is returned from DD connector which will be handled in publicapi.

### CODE REVIEW

File: src/main/java/uk/gov/pay/directdebit/payments/resources/PaymentResource.java [hunk 1]
```diff
@@ -57,6 +58,10 @@ public Response collectPaymentFromMandate(@PathParam("accountId") GatewayAccount
         LOGGER.info("Received collect payment from mandate request");
         collectPaymentRequestValidator.validate(collectPaymentRequestMap);
         Payment paymentToCollect = collectService.collect(gatewayAccount, CollectPaymentRequest.of(collectPaymentRequestMap));
+        if(!paymentService.isPaymentMandateStateValid(paymentToCollect)) {
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/directdebit/payments/resources/PaymentResource.java]: can you auto-format please

Comment (stephencdaly) [reply src/main/java/uk/gov/pay/directdebit/payments/resources/PaymentResource.java]: this should be done inside the collect service I think. we definitely shouldn't be doing it after we collect the payment.

Comment (RoryMalcolm) [reply src/main/java/uk/gov/pay/directdebit/payments/resources/PaymentResource.java]: Refactored

File: src/test/java/uk/gov/pay/directdebit/payments/resources/PaymentResourceIT.java [hunk 2]
```diff
@@ -311,6 +311,35 @@ public void shouldRetrieveAPayment_fromPublicApiEndpoint() {
...
+                .build());
+
+        String requestPath = "/v1/api/accounts/{accountId}/charges/collect"
+                .replace("{accountId}", accountExternalId);
+
+        ValidatableResponse response = givenSetup()
+                .body(postBody)
+                .post(requestPath)
+                .then()
+                .statusCode(500);
```
Comment (stephencdaly) [inline src/test/java/uk/gov/pay/directdebit/payments/resources/PaymentResourceIT.java]: should probably check the error_identifier here.

Comment (RoryMalcolm) [reply src/test/java/uk/gov/pay/directdebit/payments/resources/PaymentResourceIT.java]: Done

**Pull request merged**